internal/core/src/query/SearchBruteForce.cpp

```diff
@@ -48,15 +51,29 @@ BruteForceSearch(const dataset::SearchDataset& dataset,
             {knowhere::meta::DIM, dim},
             {knowhere::meta::TOPK, topk},
         };
+
         sub_result.mutable_seg_offsets().resize(nq * topk);
         sub_result.mutable_distances().resize(nq * topk);
 
-        auto stat =
-            knowhere::BruteForce::SearchWithBuf(base_dataset, query_dataset, sub_result.mutable_seg_offsets().data(),
-                                                sub_result.mutable_distances().data(), config, bitset);
+        if (conf.contains(RADIUS)) {
+            config[RADIUS] = conf[RADIUS];
+            if (conf.contains(RANGE_FILTER)) {
+                config[RANGE_FILTER] = conf[RANGE_FILTER];
+                CheckRangeSearchParam(config[RADIUS], config[RANGE_FILTER], dataset.metric_type);
+            }
+            auto result = SortRangeSearchResult(
+                knowhere::BruteForce::RangeSearch(base_dataset, query_dataset, config, bitset).value(), topk, nq,
+                dataset.metric_type);
+            std::copy_n(GetDatasetIDs(result), nq * topk, sub_result.get_seg_offsets());
```

Comment:
This could be done, but looks like no an easy way to reach, the interface has assumed we use `SearchResult` as return type, which store offsets in a `std::vector`, we can't simply pass the ownership of these pointers into the result. We need a better type rather than `std::vector`

Reply:
This could be done, but looks like no an easy way to reach, the interface has assumed we use SearchResult as return type, which store offsets in a std::vector, we can't simply pass the ownership of these pointers into the result. We need a better type rather than std::vector